### PR TITLE
fix UB & update `README`

### PR DESCRIPTION
**Pull request by Rudxain**
- I've just read that uninitialized vars may cause UB, lol
- Removed the `static`, for safety reasons

### CODE REVIEW

**Comment by yxor** [comment]
Any updates on this?

**Comment by Rudxain** [comment]
I got busy IRL, so I couldn't do progress. And my HDD died, so I'll have to use a basic text editor and an online compiler. When I have time, I'll try installing GCC on my Android using Termux

**Comment by Rudxain** [comment]
I came back! I'll try finishing it this month

**Comment by Rudxain** [comment]
@yxor Done! 🎉

BTW, I don't have GCC installed, so I used Clang:
```sh
clang sleep.c -o sleep -O3 -Wall -Werror
clang sort.c -o sort -O3 -Wall -Werror
```

I didn't try to compile after the style/fmt changes, because my LSP (`clangd`) didn't complain about syntax errors. I recommend compiling with GCC anyway, just in case

File: sort.c [hunk 1]
```diff
@@ -18,54 +19,57 @@ void swap_chars(char* a, char* b)
...
     {
         s[i++] = '0' + (n % 10u);
-        n /= 10u;
+        if ((n /= 10u) == 0u)
+        {
+            s[i] = '\0';
+            // convert to big-endian
+            strrev_in_place(s, i);
+            return;
+        };
```
Comment (Rudxain) [inline sort.c:48]: This could be turned into a `do ... while`

File: sort.c [hunk 2]
```diff
@@ -18,54 +19,57 @@ void swap_chars(char* a, char* b)
...
 {
-    if (n == 0u)
-        return "0";
-    u_int8_t i;
-    //`(1<<32)-1` has 10 digits.
-    //this will sometimes be a negligible "soft" memory-leak.
-    static char s[11u];
-    while (n > 0u)
+    u_int8_t i = 0u;
+    while(1)
```
Comment (Rudxain) [inline sort.c]: This could be turned into a `do ... while`

**Pull request merged**